app/src/main/java/com/trianguloy/urlchecker/modules/list/WebhookModule.java

```diff
@@ -0,0 +1,191 @@
+    
+    private final GenericPref.Str webhookUrl;
+    private final GenericPref.Bool autoSend;
+    private TextView statusText;
+    private UrlData currentUrl;
+
+    public static final List<AutomationRules.Automation<WebhookDialog>> AUTOMATIONS = new ArrayList<>();
+    static {
+        AUTOMATIONS.add(new AutomationRules.Automation<>(
+            "webhook_send",
+            R.string.mWebhook_auto_send,
+            dialog -> dialog.sendToWebhook()
+        ));
+    }
+
+    public WebhookDialog(MainDialog dialog) {
+        super(dialog);
+        webhookUrl = WebhookModule.WEBHOOK_URL_PREF(dialog);
+        autoSend = WebhookModule.AUTO_SEND_PREF(dialog);
+    }
+
+    @Override
+    public int getLayoutId() {
+        return R.layout.dialog_webhook;
+    }
+
+    @Override
+    public void onInitialize(View views) {
+        statusText = views.findViewById(R.id.status_text);
+        views.findViewById(R.id.send_button).setOnClickListener(v -> sendToWebhook());
+    }
+
+    @Override
+    public void onDisplayUrl(UrlData urlData) {
+        currentUrl = urlData;
+        if (autoSend.get()) {
+            sendToWebhook();
+        }
+    }
+
+    private void sendToWebhook() {
+        if (currentUrl == null) return;
+        
+        String webhook = webhookUrl.get();
+        if (webhook.isEmpty()) {
+            showStatus(R.string.mWebhook_no_url, true);
+            return;
+        }
+
+        showStatus(R.string.mWebhook_sending, false);
+        
+        executor.execute(() -> {
+            try {
+                JSONObject json = new JSONObject();
+                json.put("url", currentUrl.url);
+                json.put("timestamp", new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.US)
+                    .format(new Date()));
+                
+                URL url = new URL(webhook);
+                HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+                conn.setRequestMethod("POST");
+                conn.setRequestProperty("Content-Type", "application/json");
+                conn.setDoOutput(true);
+
+                try (OutputStream os = conn.getOutputStream()) {
+                    byte[] input = json.toString().getBytes(StandardCharsets.UTF_8);
+                    os.write(input, 0, input.length);
+                }
+
+                int responseCode = conn.getResponseCode();
+                if (responseCode >= 200 && responseCode < 300) {
+                    showStatus(R.string.mWebhook_success, false);
+                } else {
+                    showStatus(R.string.mWebhook_error, true);
+                }
+
+            } catch (Exception e) {
+                showStatus(R.string.mWebhook_error, true);
+            }
+        });
+    }
+
+    private void showStatus(int stringId, boolean isError) {
+        getActivity().runOnUiThread(() -> {
+            statusText.setText(stringId);
```

Comment:
You can use AndroidUtils.setRoundedColor with R.color.warning (or even R.color.error) if there was an error. (AndroidUtils.clearRoundedColor to clear it)